src/server/game/AuctionHouseBot/AuctionHouseBot.h

```diff
@@ -274,10 +277,10 @@ class AuctionBotAgent
 struct AuctionHouseBotStatusInfoPerType
 {
     uint32 ItemsCount;
-    uint32 QualityInfo[MAX_AUCTION_QUALITY];
+    std::unordered_map<AuctionQuality, uint32> QualityInfo;
 };
 
-typedef AuctionHouseBotStatusInfoPerType AuctionHouseBotStatusInfo[MAX_AUCTION_HOUSE_TYPE];
+typedef std::unordered_map<AuctionHouseType, AuctionHouseBotStatusInfoPerType> AuctionHouseBotStatusInfo;
```

Comment:
get rid of the typedef if it's not necessary, `typedef` for every map is not good (it makes code harder to read if you need to constantly reference back what exactly this container is you are dealing with)